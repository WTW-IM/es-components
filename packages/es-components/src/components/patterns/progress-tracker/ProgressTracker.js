import React from 'react';
import { withTheme } from 'styled-components';
import PropTypes from 'prop-types';
import {
  ProgressContainer,
  ProgressItem
} from './progress-tracker-subcomponents';

const getMappedSteps = (
  steps,
  onPastStepClicked,
  onFutureStepClicked,
  showNav
) => {
  let isPastStep = true;
  return steps.map((step, index) => {
    isPastStep = !(step.active || !isPastStep);
    return (
      <ProgressItem
        /* eslint-disable react/no-array-index-key */
        key={`${index}-${step.label}`}
        /* eslint-enable */
        active={step.active}
        isPastStep={isPastStep}
        numberOfSteps={steps.length}
        onPastStepClicked={() => onPastStepClicked(index)}
        canClickFutureStep={
          onFutureStepClicked !== null && onFutureStepClicked !== undefined
        }
        onFutureStepClicked={() => onFutureStepClicked(index)}
        label={step.label}
        showNav={showNav}
      />
    );
  });
};

const getIndexOfActiveStep = steps => steps.findIndex(step => step.active);

function ProgressTracker({
  steps,
  onPastStepClicked,
  onFutureStepClicked,
  showNav
}) {
  return (
    <ProgressContainer
      data-testid="progress-tracker"
      activeStepIndex={getIndexOfActiveStep(steps)}
      numberOfSteps={steps.length}
    >
      {getMappedSteps(steps, onPastStepClicked, onFutureStepClicked, showNav)}
    </ProgressContainer>
  );
}

ProgressTracker.propTypes = {
  steps: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      label: PropTypes.string
    })
  ),
  onPastStepClicked: PropTypes.func,
  onFutureStepClicked: PropTypes.func,
  showNav: PropTypes.bool
};

ProgressTracker.defaultProps = {
  steps: [],
  onPastStepClicked() {},
  onFutureStepClicked: null,
  showNav: true
};

export default withTheme(ProgressTracker);
