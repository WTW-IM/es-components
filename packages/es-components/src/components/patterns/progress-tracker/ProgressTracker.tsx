import React from 'react';
import { withTheme } from 'styled-components';
import PropTypes from 'prop-types';
import {
  ProgressContainer,
  ProgressItem
} from './progress-tracker-subcomponents';
import noop from '../../util/noop';

export interface Step {
  label?: Maybe<string>;
  active?: Maybe<boolean>;
}

export interface ProgressTrackerInternalProps {
  steps?: Step[];
  onPastStepClicked?: (index: number) => void;
  onFutureStepClicked?: (index: number) => void;
  showNav?: boolean;
}

export type ProgressTrackerProps = Override<
  JSXElementProps<'ol'>,
  ProgressTrackerInternalProps & OptionalThemeProps
>;

const getMappedSteps = ({
  steps,
  onPastStepClicked,
  onFutureStepClicked,
  showNav
}: Required<ProgressTrackerInternalProps>) => {
  let isPastStep = true;
  return steps.map((step, index) => {
    isPastStep = !(step.active || !isPastStep);
    return (
      <ProgressItem
        key={`${index}-${step.label}`}
        active={step.active}
        isPastStep={isPastStep}
        numberOfSteps={steps.length}
        onPastStepClicked={() => onPastStepClicked(index)}
        canClickFutureStep={
          onFutureStepClicked !== null && onFutureStepClicked !== undefined
        }
        label={step.label || ''}
        showNav={showNav}
      />
    );
  });
};

const getIndexOfActiveStep = (steps: Step[]) =>
  steps.findIndex(step => step.active);

const ProgressTracker = React.forwardRef<
  HTMLOListElement,
  ProgressTrackerProps
>(function ForwardedProgressTracker(
  {
    steps = [],
    onPastStepClicked = noop,
    onFutureStepClicked = noop,
    showNav = false,
    ...props
  },
  ref
) {
  return (
    <ProgressContainer
      ref={ref}
      data-testid="progress-tracker"
      activeStepIndex={getIndexOfActiveStep(steps)}
      numberOfSteps={steps.length}
      {...props}
    >
      {getMappedSteps({
        steps,
        onPastStepClicked,
        onFutureStepClicked,
        showNav
      })}
    </ProgressContainer>
  );
});

ProgressTracker.propTypes = {
  steps: PropTypes.arrayOf<Step>(
    PropTypes.shape({
      active: PropTypes.bool,
      label: PropTypes.string
    }).isRequired
  ),
  onPastStepClicked: PropTypes.func,
  onFutureStepClicked: PropTypes.func,
  showNav: PropTypes.bool
};

ProgressTracker.defaultProps = {
  steps: [],
  onPastStepClicked: noop,
  onFutureStepClicked: noop,
  showNav: true
};

export default withTheme(ProgressTracker);
