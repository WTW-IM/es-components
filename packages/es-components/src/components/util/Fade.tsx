import React from 'react';
import PropTypes from 'prop-types';
import Transition, {
  TransitionStatus,
  TimeoutProps
} from 'react-transition-group/Transition';

type Opacity = React.CSSProperties['opacity'];

interface FadeProps extends Partial<TimeoutProps<HTMLElement>> {
  duration?: number;
  opacity?: Opacity;
  children: NonNullable<React.ReactNode>;
}

const getFadeStyle = (status: TransitionStatus, finalOpacity: Opacity) => {
  switch (status) {
    case 'entering':
    case 'entered':
      return { finalOpacity };
    default:
      return { opacity: 0 };
  }
};

type AnyReactHTMLElement = React.ReactHTMLElement<HTMLElement>;

const TransitionChild: React.FC<{
  children: AnyReactHTMLElement;
  style: React.CSSProperties;
  transitionStatus: TransitionStatus;
}> = ({ children, style, transitionStatus }) => {
  return React.cloneElement(children, {
    style: {
      ...((children?.props?.style as Maybe<React.CSSProperties>) || {}),
      ...style,
      ...getFadeStyle(
        transitionStatus,
        typeof style.opacity == 'number' || Boolean(style.opacity)
          ? style.opacity
          : 1
      )
    }
  });
};

const Fade: React.FC<FadeProps> = ({
  children,
  duration,
  opacity,
  ...otherProps
}) => {
  const transitionStyles = {
    transition: `opacity ${duration || 150}ms linear`,
    opacity
  };

  const transitionProps = {
    ...otherProps,
    timeout: duration || otherProps.timeout || 150
  };

  const hasValidChildren =
    children &&
    (typeof children === 'object' || typeof children === 'function');

  return !hasValidChildren ? (
    <>{children}</>
  ) : (
    <Transition {...transitionProps}>
      {transitionStatus => {
        try {
          return (
            React.Children.only(children) && (
              <TransitionChild
                transitionStatus={transitionStatus}
                style={transitionStyles}
              >
                {children as AnyReactHTMLElement}
              </TransitionChild>
            )
          );
        } catch (e) {
          return React.Children.map(children, child => {
            return !child ? (
              child
            ) : (
              <TransitionChild
                transitionStatus={transitionStatus}
                style={transitionStyles}
              >
                {child as AnyReactHTMLElement}
              </TransitionChild>
            );
          });
        }
      }}
    </Transition>
  );
};

Fade.propTypes = {
  children: PropTypes.any.isRequired,
  duration: PropTypes.number,
  opacity: PropTypes.number
};

Fade.defaultProps = {
  duration: 150,
  opacity: 1
};

export default Fade;
