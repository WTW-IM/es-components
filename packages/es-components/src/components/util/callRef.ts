import React from 'react';

const callRef = <T, R extends React.ForwardedRef<T>>(ref: Maybe<R>, value: T) =>
  typeof ref === 'function'
    ? ref(value)
    : ref && Object.hasOwn(ref, 'current') && (ref.current = value);

export default callRef;

export function callRefs<T, R extends React.ForwardedRef<T>>(
  value: T,
  ...refs: R[]
) {
  refs.forEach(r => callRef(r, value));
}
