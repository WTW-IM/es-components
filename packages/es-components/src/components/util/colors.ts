import tinycolor, { ColorInput } from 'tinycolor2';

export function darken(color: ColorInput, percent: number) {
  return tinycolor(color).darken(percent).toHexString();
}

export function lighten(color: ColorInput, percent: number) {
  return tinycolor(color).lighten(percent).toHexString();
}

export function getTextColor(
  bgColor: ColorInput,
  darkestColor: ColorInput = 'black',
  lightestColor: ColorInput = 'white'
) {
  return tinycolor.readability(darkestColor, bgColor) > 10
    ? darkestColor
    : lightestColor;
}
