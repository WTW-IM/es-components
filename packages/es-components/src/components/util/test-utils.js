// eslint-env jest
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render } from '@testing-library/react';

export function ThemeComponent(props) {
  return <ThemeProvider theme={viaTheme} {...props} />;
}

ThemeComponent.propTypes = {
  children: PropTypes.node
};

export function renderWithTheme(component) {
  return render(<ThemeComponent>{component}</ThemeComponent>);
}
