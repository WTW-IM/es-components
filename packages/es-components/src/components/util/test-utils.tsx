import React, { Profiler } from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render, RenderResult } from '@testing-library/react';

export function ThemeComponent(
  props: Omit<React.ComponentPropsWithoutRef<typeof ThemeProvider>, 'theme'>
) {
  return <ThemeProvider {...props} theme={viaTheme} />;
}

ThemeComponent.propTypes = {
  children: PropTypes.node
};

const profilerRender = (id: string, phase: string, actualDuration: number) => {
  if (actualDuration < 100) return;
  console.warn('Render took too long!', { id, phase, actualDuration });
};

export function renderWithTheme(component: React.ReactElement): RenderResult {
  const { ...viewUtils } = render(<ThemeComponent>{component}</ThemeComponent>);
  return {
    ...viewUtils,
    rerender: (rerenderComponent: React.ReactNode) =>
      viewUtils.rerender(
        <Profiler id="ThemeRender" onRender={profilerRender}>
          <ThemeComponent>{rerenderComponent}</ThemeComponent>
        </Profiler>
      )
  };
}

export const setClientWidth = (width: number) => {
  Object.defineProperty(document.documentElement, 'clientWidth', {
    writable: true,
    configurable: true,
    value: width
  });
};

export const setClientHeight = (height: number) => {
  Object.defineProperty(document.documentElement, 'clientHeight', {
    writable: true,
    configurable: true,
    value: height
  });
};
