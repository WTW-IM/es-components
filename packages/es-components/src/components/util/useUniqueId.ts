import { useRef } from 'react';
import generateAlphaName from './generateAlphaName';

export default function useUniqueId(providedId?: Maybe<string>) {
  const idRef = useRef(providedId || generateAlphaName());
  return idRef.current;
}
