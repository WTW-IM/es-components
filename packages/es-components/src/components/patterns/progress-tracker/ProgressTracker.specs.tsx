import React from 'react';
import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../util/test-utils';
import useUniqueId from '../../util/useUniqueId';
import ProgressTracker, { Step } from './ProgressTracker';

jest.mock('../../util/useUniqueId');

function doMocks() {
  const useUnId = useUniqueId as jest.Mock;
  useUnId.mockImplementationOnce(() => '1');
  useUnId.mockImplementationOnce(() => '2');
  useUnId.mockImplementationOnce(() => '3');
}

afterEach(() => {
  jest.resetAllMocks();
});

test.each<Step[][]>([
  [
    [
      { label: 'step 1', active: true },
      { label: 'step 2', active: false },
      { label: 'step 3', active: false }
    ]
  ],
  [
    [
      { label: 'step 1', active: false },
      { label: 'step 2', active: true },
      { label: 'step 3', active: false }
    ]
  ],
  [
    [
      { label: 'step 1', active: false },
      { label: 'step 2', active: false },
      { label: 'step 3', active: true }
    ]
  ]
])(`ProgressTracker renders as expected`, async steps => {
  doMocks();
  renderWithTheme(<ProgressTracker steps={steps} />);
  expect(await screen.findByTestId('progress-tracker')).toMatchSnapshot();
});
