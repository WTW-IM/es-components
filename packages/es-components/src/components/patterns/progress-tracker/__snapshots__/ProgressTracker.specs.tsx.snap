// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressTracker renders as expected 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-image: linear-gradient( to right, #444 0%, #444 0%, #c5c5c5 0%, #c5c5c5 100% );
  background-size: 100% 3px;
  background-position: 0 21px;
  background-repeat: no-repeat;
  counter-reset: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 0;
  padding-left: 0;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  counter-increment: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-family: 'Source Sans Pro',sans-serif;
  margin-top: 12.5px;
  padding: 0;
  text-align: center;
}

.c2:active,
.c2:hover,
.c2:focus {
  outline: none;
}

.c2 span {
  color: #444;
  display: none;
  font-size: 18px;
  line-height: 1.1em;
}

.c2::before {
  background-color: white;
  border: 1px solid #c5c5c5;
  border-radius: 20px;
  box-shadow: 0 0 0 3px white;
  box-sizing: border-box;
  color: #444;
  content: counter(progress-tracker-counter);
  display: block;
  font-size: 16px;
  height: 20px;
  line-height: 1em;
  margin-bottom: 5px;
  text-align: center;
  width: 20px;
  position: relative;
}

.c4 {
  cursor: pointer;
}

.c4:hover::before {
  box-shadow: rgba(0,0,0,0.35) 0 0 3px 1px;
}

.c4:active span,
.c4:hover span,
.c4:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  margin-top: 0;
}

.c3::before {
  border: 5px solid #c110a0;
  border-radius: 45px;
  color: #c110a0;
  font-size: 21.96px;
  font-weight: bold;
  height: 45px;
  line-height: 1.5em;
  width: 45px;
}

.c3 span {
  color: #c110a0;
  display: inline;
  font-weight: bold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  max-width: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:first-child,
.c1:first-child button {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c1:last-child,
.c1:last-child button {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  text-align: right;
}

@media (min-width:768px) {
  .c0 {
    background-position: center 19px;
    background-size: 66.66666666666667% 3px;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 18px;
    margin-top: 0;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c2 span {
    display: inline;
    font-weight: normal;
  }
}

@media (min-width:768px) {
  .c2::before {
    border-radius: 41px;
    border-width: 3px;
    font-size: 20.008px;
    font-weight: normal;
    height: 41px;
    line-height: 1.7em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c3::before {
    border-width: 5px;
    font-size: 20.008px;
    height: 41px;
    line-height: 1.5em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c3 span {
    font-weight: bold;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 33.33333333333333%;
  }

  .c1:first-child,
  .c1:last-child,
  .c1:first-child button,
  .c1:last-child button {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    text-align: center;
  }
}

<ol
  class="c0"
  data-testid="progress-tracker"
>
  <li
    class="c1"
  >
    <button
      aria-labelledby="1-span"
      class="c2 c3"
      id="1"
    >
      <span
        id="1-span"
      >
        step 1
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="2-span"
      class="c2 c4"
      id="2"
    >
      <span
        id="2-span"
      >
        step 2
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="3-span"
      class="c2 c4"
      id="3"
    >
      <span
        id="3-span"
      >
        step 3
      </span>
    </button>
  </li>
</ol>
`;

exports[`ProgressTracker renders as expected 2`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-image: linear-gradient( to right, #444 0%, #444 50%, #c5c5c5 50%, #c5c5c5 100% );
  background-size: 100% 3px;
  background-position: 0 21px;
  background-repeat: no-repeat;
  counter-reset: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 0;
  padding-left: 0;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  counter-increment: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-family: 'Source Sans Pro',sans-serif;
  margin-top: 12.5px;
  padding: 0;
  text-align: center;
}

.c2:active,
.c2:hover,
.c2:focus {
  outline: none;
}

.c2 span {
  color: #444;
  display: none;
  font-size: 18px;
  line-height: 1.1em;
}

.c2::before {
  background-color: white;
  border: 1px solid #c5c5c5;
  border-radius: 20px;
  box-shadow: 0 0 0 3px white;
  box-sizing: border-box;
  color: #444;
  content: counter(progress-tracker-counter);
  display: block;
  font-size: 16px;
  height: 20px;
  line-height: 1em;
  margin-bottom: 5px;
  text-align: center;
  width: 20px;
  position: relative;
}

.c5 {
  cursor: pointer;
}

.c5:hover::before {
  box-shadow: rgba(0,0,0,0.35) 0 0 3px 1px;
}

.c5:active span,
.c5:hover span,
.c5:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  cursor: pointer;
}

.c3:hover::before {
  box-shadow: rgba(0,0,0,0.35) 0 0 3px 1px;
}

.c3:active span,
.c3:hover span,
.c3:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus::before {
  box-shadow: #83bffc 0 0 4px 2px;
}

.c3:focus span {
  outline: 1px dotted #83bffc;
}

.c3::before {
  border-color: #444;
}

.c4 {
  margin-top: 0;
}

.c4::before {
  border: 5px solid #c110a0;
  border-radius: 45px;
  color: #c110a0;
  font-size: 21.96px;
  font-weight: bold;
  height: 45px;
  line-height: 1.5em;
  width: 45px;
}

.c4 span {
  color: #c110a0;
  display: inline;
  font-weight: bold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  max-width: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:first-child,
.c1:first-child button {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c1:last-child,
.c1:last-child button {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  text-align: right;
}

@media (min-width:768px) {
  .c0 {
    background-position: center 19px;
    background-size: 66.66666666666667% 3px;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 18px;
    margin-top: 0;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c2 span {
    display: inline;
    font-weight: normal;
  }
}

@media (min-width:768px) {
  .c2::before {
    border-radius: 41px;
    border-width: 3px;
    font-size: 20.008px;
    font-weight: normal;
    height: 41px;
    line-height: 1.7em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c3::before {
    border-width: 5px;
    font-weight: bold;
    line-height: 1.5em;
  }
}

@media (min-width:768px) {
  .c3 span {
    font-weight: bold;
  }
}

@media (min-width:768px) {
  .c4::before {
    border-width: 5px;
    font-size: 20.008px;
    height: 41px;
    line-height: 1.5em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c4 span {
    font-weight: bold;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 33.33333333333333%;
  }

  .c1:first-child,
  .c1:last-child,
  .c1:first-child button,
  .c1:last-child button {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    text-align: center;
  }
}

<ol
  class="c0"
  data-testid="progress-tracker"
>
  <li
    class="c1"
  >
    <button
      aria-labelledby="1-span"
      class="c2 c3"
      id="1"
    >
      <span
        id="1-span"
      >
        step 1
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="2-span"
      class="c2 c4"
      id="2"
    >
      <span
        id="2-span"
      >
        step 2
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="3-span"
      class="c2 c5"
      id="3"
    >
      <span
        id="3-span"
      >
        step 3
      </span>
    </button>
  </li>
</ol>
`;

exports[`ProgressTracker renders as expected 3`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-image: linear-gradient( to right, #444 0%, #444 100%, #c5c5c5 100%, #c5c5c5 100% );
  background-size: 100% 3px;
  background-position: 0 21px;
  background-repeat: no-repeat;
  counter-reset: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 0;
  padding-left: 0;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  counter-increment: progress-tracker-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-family: 'Source Sans Pro',sans-serif;
  margin-top: 12.5px;
  padding: 0;
  text-align: center;
}

.c2:active,
.c2:hover,
.c2:focus {
  outline: none;
}

.c2 span {
  color: #444;
  display: none;
  font-size: 18px;
  line-height: 1.1em;
}

.c2::before {
  background-color: white;
  border: 1px solid #c5c5c5;
  border-radius: 20px;
  box-shadow: 0 0 0 3px white;
  box-sizing: border-box;
  color: #444;
  content: counter(progress-tracker-counter);
  display: block;
  font-size: 16px;
  height: 20px;
  line-height: 1em;
  margin-bottom: 5px;
  text-align: center;
  width: 20px;
  position: relative;
}

.c3 {
  cursor: pointer;
}

.c3:hover::before {
  box-shadow: rgba(0,0,0,0.35) 0 0 3px 1px;
}

.c3:active span,
.c3:hover span,
.c3:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus::before {
  box-shadow: #83bffc 0 0 4px 2px;
}

.c3:focus span {
  outline: 1px dotted #83bffc;
}

.c3::before {
  border-color: #444;
}

.c4 {
  margin-top: 0;
}

.c4::before {
  border: 5px solid #c110a0;
  border-radius: 45px;
  color: #c110a0;
  font-size: 21.96px;
  font-weight: bold;
  height: 45px;
  line-height: 1.5em;
  width: 45px;
}

.c4 span {
  color: #c110a0;
  display: inline;
  font-weight: bold;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  max-width: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:first-child,
.c1:first-child button {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c1:last-child,
.c1:last-child button {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  text-align: right;
}

@media (min-width:768px) {
  .c0 {
    background-position: center 19px;
    background-size: 66.66666666666667% 3px;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 18px;
    margin-top: 0;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c2 span {
    display: inline;
    font-weight: normal;
  }
}

@media (min-width:768px) {
  .c2::before {
    border-radius: 41px;
    border-width: 3px;
    font-size: 20.008px;
    font-weight: normal;
    height: 41px;
    line-height: 1.7em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c3::before {
    border-width: 5px;
    font-weight: bold;
    line-height: 1.5em;
  }
}

@media (min-width:768px) {
  .c3 span {
    font-weight: bold;
  }
}

@media (min-width:768px) {
  .c4::before {
    border-width: 5px;
    font-size: 20.008px;
    height: 41px;
    line-height: 1.5em;
    width: 41px;
  }
}

@media (min-width:768px) {
  .c4 span {
    font-weight: bold;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 33.33333333333333%;
  }

  .c1:first-child,
  .c1:last-child,
  .c1:first-child button,
  .c1:last-child button {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    text-align: center;
  }
}

<ol
  class="c0"
  data-testid="progress-tracker"
>
  <li
    class="c1"
  >
    <button
      aria-labelledby="1-span"
      class="c2 c3"
      id="1"
    >
      <span
        id="1-span"
      >
        step 1
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="2-span"
      class="c2 c3"
      id="2"
    >
      <span
        id="2-span"
      >
        step 2
      </span>
    </button>
  </li>
  <li
    class="c1"
  >
    <button
      aria-labelledby="3-span"
      class="c2 c4"
      id="3"
    >
      <span
        id="3-span"
      >
        step 3
      </span>
    </button>
  </li>
</ol>
`;
